include/zephyr/init.h

```diff
@@ -135,13 +137,21 @@ struct init_entry {
 	__attribute__((__section__(                                           \
 		".z_init_" #level STRINGIFY(prio)"_" STRINGIFY(sub_prio)"_")))
 
+/**
+ * @brief Issue a warning if the given init level is deprecated.
+ *
+ * @param level Init level
+ */
+#define Z_INIT_CHECK_DEPRECATED_LEVEL(level)                                   \
+	COND_CODE_1(Z_INIT_SMP_##level, (__WARN("SMP level is deprecated")), ())
```

Comment:
Suggestion for GCC 11. Would this work?

```c
#define Z_INIT_DEPRECATED_EARLY
#define Z_INIT_DEPRECATED_PRE_KERNEL_1
#define Z_INIT_DEPRECATED_PRE_KERNEL_2
#define Z_INIT_DEPRECATED_POST_KERNEL
#define Z_INIT_DEPRECATED_APPLICATION
#define Z_INIT_DEPRECATED_SMP		__WARN("SMP level is deprecated")
```
```suggestion
#define Z_INIT_CHECK_DEPRECATED_LEVEL(level) Z_INIT_DEPRECATED_##level
```

Reply:
great idea, seems to work! thanks, that saved a lot of investigation time :-)